modules/core/src/main/scala/scala/build/Os.scala

```diff
@@ -8,4 +8,6 @@ object Os {
       os.Path(os.pwd.toIO.getCanonicalFile)
     else
       os.pwd
+  def isArmArchitecture: Boolean =
+    os.proc("uname", "-p").call(cwd = os.pwd).out.trim() == "arm"
```

Comment:
Wouldn't it work by checking some Java property (like `"os.arch"`)? It seems this is going to make even the x86_64 version of Scala CLI run or load M1 binaries, which might not work when loading libraries, but also may not be what users expect also (the x86_64 version should run and load x86_64 too under the hood, I'd say).